src/main/scala/com.snowplowanalytics.sauna/apis/UrbanAirship.scala

```diff
@@ -0,0 +1,176 @@
+import scala.io.Source.fromInputStream
+import scala.io.Source.fromFile
+import scala.util.{ Failure, Success }
+import scala.collection.mutable.{ Map => MutableMap }
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+import akka.actor.ActorRef
+ 
+import play.api.Play.current
+import play.api.libs.ws._
+import play.api.libs.ws.ning.NingAsyncHttpClientConfigBuilder
+
+import responders.urbanairship.UAResponder.Airship
+
+import play.api.libs.json.Json
+
+//import play.api.mvc._
+//import play.api.Play.current
+//import play.api.libs.ws._
+
+import loggers.Logger.Notification
+
+class urbanAirship(implicit logger: ActorRef) {
+  import UrbanAirship._
+
+  /**
+   * Converts the gouped data in the form of a map into upload requests to UrbanAirship and checks the status of the operation
+   *
+   * @param is the map of data grouped by appKeys and then by listNames
+   */
+
+  def maptoRequest(appMap: Map[String, Map[String, List[Airship]]]): Unit =
+    {
+      
+      var listNamesToKeyMap = Map[String,(Boolean,String)]()
+      for ((appKey, value) <- appMap) {
+
+        for ((listName, valueList) <- value) {
+          
+          val (status,userPass) = makeRequest(appKey, listName, valueList)
+          listNamesToKeyMap += (listName -> (status,userPass))
+
+        }
+
+      }
+
+      val timeout = System.currentTimeMillis() + 15 * 60 * 1000
+
+      while (listNamesToKeyMap.size > 0 && timeout >= System.currentTimeMillis()) {
+
+        for ((listName, requestTuple) <- listNamesToKeyMap) {
+        
+            val (requestAccepted,userPass) = requestTuple
+            if(requestAccepted == true)
+            {
+              val status = checkStatus(listName, userPass)
+              if (status == "ready") {
+              listNamesToKeyMap -= listName
+              }
+            }
+            else
+            {
+               listNamesToKeyMap -= listName
+            }
+        
+        }
+
+      }
+
+      if (timeout >= System.currentTimeMillis()) {
+
+        for ((listName, userpass) <- listNamesToKeyMap) {
+          logger ! Notification("listName " + listName + " upload failed")
+        }
+
+      }
+      logger ! Notification("Upload Complete !")
+    }
+
+  /**
+   * Converts the gouped data in the form of a map into upload requests to UrbanAirship and checks the status of the operation
+   *
+   * @param appKey is the Application Key of the App in UrbanAirship
+   * @param listName is the name for the list to which the uploads are directed
+   * @param valueMap is the List of Identifier and IdentifierTypes for the upload
+   * @return userpass is the authentication key which is the combination of Appkey and MasterKey needed to make the request
+   */
+  def makeRequest(appKey: String, listName: String, values: List[Airship]): (Boolean,String) =
+    {
+  
+      val client = {
```

Comment:
By the way, @rmanojcit! Not related to our ongoing refactoring, but:
+ We already have [ws-play](https://github.com/snowplow/sauna/pull/35/files#diff-40718f6f11d36e6fec23a9f88b986697R35) in dependencies. No need to include it twice.
+ You can access it by `utils.wsClient` in global scope
+ I don't think there's a need to release all resources (close) after each HTTP request